registry/config.go

```diff
@@ -240,15 +240,28 @@ func validateNoSchema(reposName string) error {
 
 // ValidateRepositoryName validates a repository name
 func ValidateRepositoryName(reposName string) error {
-	var err error
-	if err = validateNoSchema(reposName); err != nil {
-		return err
+	_, _, err := loadRepositoryName(reposName, true)
```

Comment:
Nit: ValidateRepositoryName is now a thin wrapper over loadRepositoryName. Given that, its better to give them similar names for better readability.